VideoLocker/src/main/java/org/edx/mobile/model/api/CourseEntry.java

```diff
@@ -248,22 +252,79 @@ public boolean isGroupAvailable(SocialUtils.SocialType type) {
 
     }
 
-    public String getDescription(Context context){
+    public String getDescription(Context context, boolean withStartDate) {
         StringBuilder detailBuilder = new StringBuilder();
-        if ( getOrg() != null){
-            detailBuilder.append( getOrg());
+
+        if (!TextUtils.isEmpty(org)) {
+            detailBuilder.append(org);
         }
-        if ( getNumber() != null) {
-            if (detailBuilder.length() > 0){
+
+        if (!TextUtils.isEmpty(number)) {
+            if (detailBuilder.length() > 0) {
                 detailBuilder.append(" | ");
             }
-            detailBuilder.append( getNumber());
+            detailBuilder.append(number);
 
         }
-        //https://openedx.atlassian.net/browse/MA-858
-        //we remove the ending data for now
+
+        if (withStartDate) {
+            if (detailBuilder.length() > 0) {
+                detailBuilder.append(" | ");
+            }
+
+            detailBuilder.append(getFormattedStartDate(context));
+        }
+
+        if (detailBuilder.length() > 0) {
+            int secondlastCharIndex = detailBuilder.length() - 2;
+            if (detailBuilder.charAt(secondlastCharIndex) == '|') {
+                detailBuilder.delete(secondlastCharIndex, detailBuilder.length());
```

Comment:
Are we just trying to join some strings with pipe characters?

It would be much simpler to build a `List<String>` and use TextUtils.join(" | ", list)